python_modules/dagster/dagster/_core/definitions/asset_reconciliation_sensor.py

```diff
@@ -48,6 +50,113 @@
     from dagster._utils.caching_instance_queryer import CachingInstanceQueryer  # expensive import
 
 
+class AutoMaterializeCondition(Enum):
+    """Represents the set of conditions that can trigger auto-materialization for an asset."""
+
+    FRESHNESS = "FRESHNESS"
+    DOWNSTREAM_FRESHNESS = "DOWNSTREAM_FRESHNESS"
+    PARENT_MATERIALIZED = "PARENT_MATERIALIZED"
+    MISSING = "MISSING"
+
+
+@whitelist_for_serdes
+class AutoMaterializeReason(NamedTuple):
+    """Denotes the reason that the auto-materialize logic decided that an asset should be materialized.
+
+    In the future, may be extended to support additional details attached to raw condition.
+
+    Should not be instantiated directly by the user.
+    """
+
+    condition: AutoMaterializeCondition
+
+    @staticmethod
+    def freshness() -> "AutoMaterializeReason":
+        return AutoMaterializeReason(condition=AutoMaterializeCondition.FRESHNESS)
+
+    @staticmethod
+    def downstream_freshness() -> "AutoMaterializeReason":
+        return AutoMaterializeReason(condition=AutoMaterializeCondition.DOWNSTREAM_FRESHNESS)
+
+    @staticmethod
+    def parent_materialized() -> "AutoMaterializeReason":
+        return AutoMaterializeReason(condition=AutoMaterializeCondition.PARENT_MATERIALIZED)
+
+    @staticmethod
+    def missing() -> "AutoMaterializeReason":
+        return AutoMaterializeReason(condition=AutoMaterializeCondition.MISSING)
+
+
+class AutoMaterializeSkipCondition(Enum):
+    """Represents the set of conditions that can prevent an asset from being auto-materialized."""
+
+    PARENT_OUTDATED = "PARENT_OUTDATED"
+
+
+@whitelist_for_serdes
+class AutoMaterializeSkipReason(NamedTuple):
+    """Denotes the reason that the auto-materialize logic decided that an asset should not be
+    materialized.
+
+    In the future, may be extended to support additional details attached to raw condition.
+
+    Should not be instantiated directly by the user.
+    """
+
+    condition: AutoMaterializeSkipCondition
+
+    @staticmethod
+    def parent_outdated() -> "AutoMaterializeSkipReason":
+        return AutoMaterializeSkipReason(condition=AutoMaterializeSkipCondition.PARENT_OUTDATED)
+
+
+@whitelist_for_serdes
+class AutoMaterializeEvaluation(NamedTuple):
```

Comment:
Docstring plz

Reply:
Also, should this have an `AssetKey` attribute?

Reply:
good calls